forge/db/migrations/20221208-01-add-state-to-subscription.js

```diff
@@ -0,0 +1,20 @@
+/**
+ * Add status to subscriptions
+ */
+
+const { DataTypes } = require('sequelize')
+
+/** @type {import('sequelize-cli').Migration} */
+module.exports = {
+    async up (queryInterface, Sequelize) {
+        await queryInterface.addColumn('Subscriptions', 'status', {
+            type: DataTypes.ENUM,
+            values: ['active', 'canceled'], // Full list from stripe: trialing,active,incomplete,incomplete_expired,past_due,canceled,unpaid
+            defaultValue: 'active'
+        })
+    },
+
+    async down (queryInterface, Sequelize) {
+        await queryInterface.removeColumn('Subscriptions', 'status')
+    }
+}
```

Comment:
Should this be in `forge/ee/db/migrations` since it should only be applied if there is a license and billing is enabled

Reply:
We do not currently have any concept of ee-only migrations - because a license could get applied at any time to enable the EE components.

The issue here is the migration will get applied the first time FF runs and it will expect to find the Subscriptions table to exist - which would only be true if it had previously run with a license.

When billing was introduced, it looks like no migration was added to create the Subscriptions table - so it is left to Sequelize to do it on demand. Which means the table won't have been created by the time this migration runs on a clean system.

This migration needs to check to see if the Subscriptions table exists and only then attempt to add the column.

Reply:
In [b65b0a6](https://github.com/flowforge/flowforge/pull/1417/commits/b65b0a6c02de65a59a2e2b0436686162e3b77686) I've added a guard locally and verified it works by deleting the subscription table and migration from the meta versions.